### DISCUSSION TITLE
TFA pool thermometer 30.3240.10

### DISCUSSION THREAD

**Discussion by mikaelsundin** [merbanan/rtl_433#2651] [General]
I have started to investigate TFA pool thermometer 30.3240.10.

What i found so far.
The protocol seems to be 36bits payload and repeat 4 times every transmission.

`
SIIIIIIIIXXXX00CCTTTTTTTTTTTT00000000S

I = probably preamble, but can be a ID (someone else need to verify this with another unit)
If the NTC sensor is missing this field changes to S10011100...

XXXX set at startup and seems to be based on selected Channel. (will not change after startup if switching Channel)

CC = Selected Channel
T = 12bit signed integer with scale of 0.1°.

Example Data CH1
S100100000111000011111110110000000000S = -2
S100100000001000000001000101100000000S = 13.9
S100100000001000000001001110100000000S = 15.7
S100100000111000000001110101100000000S = 23.5

Example Data CH1
S100100000111000100001110101100000000S = 23.5
S100100001101000100010000010000000000S = 26.0

The radio protokoll seems to match other TFA 433Mhz sensors.
``     __   0   __     1     __    START/STOP    __  ``
``   _|  |_____|  |_________|  |________________|  |_``

0: 1500 - 2500 us
1: 3500 - 4500 us
S: 7500 - 9000 us

**Comment by mikaelsundin** [comment]
Arduino code to decode packets
https://github.com/mikaelsundin/homeautomation_tfa_pool_thermometer

**Comment by gdt** [comment]
It seems you could look at src/devices/tfa* and see if one of them can be adapted to also decode this.  (I am assuming the point is to add support to rtl_433!).

**Reply by mikaelsundin** [reply]
This seems to match the timing. 
https://github.com/merbanan/rtl_433/blob/master/src/devices/tfa_pool_thermometer.c

Unsure if correct way is to extend same decoder to accept both 28bits and 36bits.
Both tfa_pool_thermometer.c and 30.3240.10 is a pool thermometer from TFA.

**Reply by merbanan** [reply]
The protocol seems to be different. Implement in its own file in that case. It looks like there is no checksum.

**Comment by merbanan** [comment]
Might match the nexus protocol or prologue.

**Reply by MD1848** [reply]
Hi ,
I have the same thermometer and had no success with protocol 3 and 19.
Is there a solution in the meantime? Unfortunately, I have no programming knowledge.

**Reply by zuckschwerdt** [reply]
Can you grab a sample and upload here as zip to verify the timings? https://triq.org/rtl_433/ANALYZE.html

This flex decoder should work (unclear on packet gaps), can you get data with it? If that works try to reset (battery out/in) a couple of times and record data each time.
`rtl_433 -R 0 -X 'n=name,m=OOK_PPM,s=2000,l=4000,y=8000,g=8000,r=10000,bits>=36'`

**Reply by MD1848** [reply]
output:
@gastonia:~$ rtl_433 -R 0 -X 'n=name,m=OOK_PPM,s=2000,l=4000,y=8000,g=8000,r=10000,bits>=36'
rtl_433 version 23.11-115-g8be3ef15 branch master at 202405022024 inputs file rtl_tcp RTL-SDR SoapySDR with TLS
Reading conf from "rtl_433.conf".

New defaults active, use "-Y classic -s 250k" if you need the old defaults

newmodel option (-M) is deprecated.
Disabling all device decoders.
Found Rafael Micro R820T tuner
[SDR] Using device 0: Realtek, RTL2838UHIDIR, SN: 00000001, "Generic RTL2832U OEM"
Exact sample rate is: 1000000.026491 Hz
[R82XX] PLL not locked!
Allocating 15 zero-copy buffers
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-05-05 18:05:33
model     : name         count     : 1             num_rows  : 1             rows      :
len       : 146          data      : 0000000000000000000000000000000000000
codes     : {146}0000000000000000000000000000000000000
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-05-05 18:06:01
model     : name         count     : 1             num_rows  : 1             rows      :
len       : 43           data      : 00000000000
codes     : {43}00000000000
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-05-05 18:06:01
model     : name         count     : 1             num_rows  : 1             rows      :
len       : 43           data      : 00000000000
codes     : {43}00000000000
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-05-05 18:06:01
model     : name         count     : 1             num_rows  : 1             rows      :
len       : 43           data      : 00000000000
codes     : {43}00000000000
^CSignal caught, exiting!
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-05-05 18:06:04                    enabled   : 1             since     : 2024-05-05T18:05:24                     frames    :
count     : 21           fsk       : 0             events    : 4             stats     :
device    : 0            name      : General purpose decoder 'name'          events    : 6             ok        : 4             messages  : 4
abort_length: 2

the zip file:
[rtl_tmp.gz](https://github.com/merbanan/rtl_433/files/15214153/rtl_tmp.gz)

**Reply by zuckschwerdt** [reply]
Those zeros don't look good, that's only noise.

All files except one are empty (noise only), and that one file contains FSK data, presumably a different sensor.

Try a SDR UI and in the waterfall display locate the frequency and confirm a good reception first.

**Reply by MD1848** [reply]
yes, I have a few others that work:
...
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-05-07 12:33:28                    Protocol  : 20
model     : Ambientweather-F007TH                  House Code: 227
Channel   : 2            Battery   : 1             Temperature: 19.9 C       Humidity  : 68 %          Integrity : CRC
Modulation: ASK          Freq      : 433.9 MHz
RSSI      : -0.1 dB      SNR       : 19.9 dB       Noise     : -20.0 dB
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-05-07 12:33:57                    Protocol  : 150
model     : Klimalogg-Pro Id        : 589b
Battery   : 1            Temperature: 17.1 C       Humidity  : 106           Sequence Number: 3        Integrity : CRC
Modulation: ASK          Freq      : 868.3 MHz
RSSI      : -12.1 dB     SNR       : 15.4 dB       Noise     : -27.5 dB
...

But I sent the pool transmitter back to A...n

**Reply by wtaferner** [reply]
> Can you grab a sample and upload here as zip to verify the timings? https://triq.org/rtl_433/ANALYZE.html
> 
> This flex decoder should work (unclear on packet gaps), can you get data with it? If that works try to reset (battery out/in) a couple of times and record data each time. `rtl_433 -R 0 -X 'n=name,m=OOK_PPM,s=2000,l=4000,y=8000,g=8000,r=10000,bits>=36'`

Ok, not sure about the upload, but I have done the manual copy & paste with reference temperature and the data seems to have changed accordingly...let me know what I could share additionally 😄  

```bash
time      : 2024-08-02 17:02:18 -> temperature: 29,1°C
model     : name         count     : 7             num_rows  : 8             rows      : 
len       : 0            data      : , 
len       : 37           data      : 9020123000, 
len       : 37           data      : 9020123000, 
len       : 37           data      : 9020123000, 
len       : 37           data      : 9020123000, 
len       : 37           data      : 9020123000, 
len       : 37           data      : 9020123000, 
len       : 37           data      : 9020123000
codes     : {0}0, {37}9020123000, {37}9020123000, {37}9020123000, {37}9020123000, {37}9020123000, {37}9020123000, {37}9020123000

time      : 2024-08-02 16:55:25 -> temperature: 29,0°C
model     : name         count     : 7             num_rows  : 8             rows      : 
len       : 0            data      : , 
len       : 37           data      : 9010122000, 
len       : 37           data      : 9010122000, 
len       : 37           data      : 9010122000, 
len       : 37           data      : 9010122000, 
len       : 37           data      : 9010122000, 
len       : 37           data      : 9010122000, 
len       : 37           data      : 9010122000
codes     : {0}0, {37}9010122000, {37}9010122000, {37}9010122000, {37}9010122000, {37}9010122000, {37}9010122000, {37}9010122000

time      : 2024-08-02 16:53:50 -> temperature: 28,8°C
model     : name         count     : 7             num_rows  : 8             rows      : 
len       : 0            data      : , 
len       : 37           data      : 9000120000, 
len       : 37           data      : 9000120000, 
len       : 37           data      : 9000120000, 
len       : 37           data      : 9000120000, 
len       : 37           data      : 9000120000, 
len       : 37           data      : 9000120000, 
len       : 37           data      : 9000120000
codes     : {0}0, {37}9000120000, {37}9000120000, {37}9000120000, {37}9000120000, {37}9000120000, {37}9000120000, {37}9000120000


time      : 2024-08-02 16:54:19 -> temperature: 18,8°C
model     : name         count     : 7             num_rows  : 8             rows      : 
len       : 0            data      : , 
len       : 37           data      : 90300bc000, 
len       : 37           data      : 90300bc000, 
len       : 37           data      : 90300bc000, 
len       : 37           data      : 90300bc000, 
len       : 37           data      : 90300bc000, 
len       : 37           data      : 90300bc000, 
len       : 37           data      : 90300bc000
codes     : {0}0, {37}90300bc000, {37}90300bc000, {37}90300bc000, {37}90300bc000, {37}90300bc000, {37}90300bc000, {37}90300bc000

```

**Reply by zuckschwerdt** [reply]
Thanks! If we check the data [in a BitBench](https://triq.net/bitbench#c=%7B37%7D9010122000%5B29%2C0%C2%B0C%5D&c=%7B37%7D9000120000%5B28%2C8%C2%B0C%5D&c=%7B37%7D90300bc000%5B18%2C8%C2%B0C%5D&f=S%3Fb%20ID%3F8h%20START%3F4h%202b%20CH%3A2h%20TEMP%3F12d%20%3F8h%20S%3Fb&z=1&cw=4) it seems shifted by one bit (the temperature does not step in 0.1 C increments).

It appears the start/stop bits are not properly decoded. Can you run `rtl_433 -S unknown` to get a sample, then zip and upload here (drag'n'drop)? That way we can fine tune the timings to match better.

**Reply by wtaferner** [reply]
And to have some weird stuff as well...

```bash

time      : 2024-08-02 17:15:55 -> temperature: -47,3°C
model     : name         count     : 7             num_rows  : 8             rows      : 
len       : 0            data      : , 
len       : 37           data      : 9240e27000, 
len       : 37           data      : 9240e27000, 
len       : 37           data      : 9240e27000, 
len       : 37           data      : 9240e27000, 
len       : 37           data      : 9240e27000, 
len       : 37           data      : 9240e27000, 
len       : 37           data      : 9240e27000
codes     : {0}0, {37}9240e27000, {37}9240e27000, {37}9240e27000, {37}9240e27000, {37}9240e27000, {37}9240e27000, {37}9240e27000

```

**Reply by wtaferner** [reply]
```bash
rtl_433 version 23.11 (2023-11-28) inputs file rtl_tcp RTL-SDR SoapySDR
Detached kernel driver
Found Rafael Micro R820T tuner
[SDR] Using device 0: Realtek, RTL2838UHIDIR, SN: 00000001, "Generic RTL2832U OEM"
Exact sample rate is: 250000.000414 Hz
[R82XX] PLL not locked!
Allocating 15 zero-copy buffers
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:22:23
model     : Prologue-TH  subtype   : 9             id        : 13
Channel   : 1            Battery   : 0             Temperature: 30.00 C      Humidity  : 0 %
Button    : 0
*** Saving signal to file g001_433.92M_250k.cu8 (69756 samples, 262144 bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:22:43
model     : Prologue-TH  subtype   : 9             id        : 1
Channel   : 1            Battery   : 0             Temperature: 30.00 C      Humidity  : 0 %
Button    : 0
*** Saving signal to file g002_433.92M_250k.cu8 (204787 samples, 524288 bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:22:54
model     : Prologue-TH  subtype   : 9             id        : 13
Channel   : 1            Battery   : 0             Temperature: 30.00 C      Humidity  : 0 %
Button    : 0
*** Saving signal to file g003_433.92M_250k.cu8 (69215 samples, 262144 bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:23:04
model     : Prologue-TH  subtype   : 9             id        : 1
Channel   : 1            Battery   : 0             Temperature: 30.00 C      Humidity  : 0 %
Button    : 0
*** Saving signal to file g004_433.92M_250k.cu8 (201019 samples, 524288 bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:23:12
model     : Prologue-TH  subtype   : 9             id        : 2
Channel   : 1            Battery   : 0             Temperature: 30.00 C      Humidity  : 0 %
Button    : 0
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:23:21
model     : Prologue-TH  subtype   : 9             id        : 5
Channel   : 1            Battery   : 0             Temperature: 30.10 C      Humidity  : 0 %
Button    : 0
*** Saving signal to file g005_433.92M_250k.cu8 (67119 samples, 262144 bytes)
*** Saving signal to file g006_433.92M_250k.cu8 (323315 samples, 655360 bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:23:38
model     : Prologue-TH  subtype   : 9             id        : 1
Channel   : 1            Battery   : 0             Temperature: 30.20 C      Humidity  : 0 %
Button    : 0
*** Saving signal to file g007_433.92M_250k.cu8 (199740 samples, 524288 bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:23:47
model     : Prologue-TH  subtype   : 9             id        : 0
Channel   : 1            Battery   : 0             Temperature: 30.30 C      Humidity  : 0 %
Button    : 0



*** Saving signal to file g008_433.92M_250k.cu8 (70764 samples, 262144 bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
time      : 2024-08-02 17:24:09
model     : Prologue-TH  subtype   : 9             id        : 14
Channel   : 1            Battery   : 0             Temperature: 30.40 C      Humidity  : 0 %
Button    : 0
```

[sample_series.zip](https://github.com/user-attachments/files/16472336/sample_series.zip)

Ok, I think, this one should be better...

**Reply by wtaferner** [reply]
Hmm, actually it works already I realized...it is exactly the same as Prologue-TH and has no battery, no humidity sensor and also no button...it will change the ID on every reset / battery change.

**Comment by wtaferner** [comment]
How far are we away to decode this one?
I just bought one and I am eager to contribute the data samples or at least give it a try.

**Reply by zuckschwerdt** [reply]
We need https://github.com/merbanan/rtl_433/discussions/2651#discussioncomment-9319999
